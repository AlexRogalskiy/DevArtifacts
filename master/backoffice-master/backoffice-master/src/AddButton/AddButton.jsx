import React from 'react'
import PropTypes from 'prop-types'

import {
  Button,
  withStyles,
} from '@material-ui/core'
import AddIcon from '@material-ui/icons/Add'

const styles = theme => ({
  button: {
    position: 'fixed',
    right: theme.spacing.unit * 3,
    bottom: theme.spacing.unit * 3,
  },
})

const AddButton = ({ onClick, classes }) => (
  <Button
    variant="fab"
    color="secondary"
    aria-label="add"
    className={classes.button}
    onClick={onClick}
  >
    <AddIcon />
  </Button>
)

AddButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
}

export default withStyles(styles)(AddButton)
