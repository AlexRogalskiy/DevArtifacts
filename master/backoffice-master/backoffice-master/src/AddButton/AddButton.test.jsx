import React from 'react'
import { mount } from 'enzyme'

import AddButton from './'

it('renders correctly', () => {
  const tree = mount(<AddButton onClick={() => {}} />)

  expect(tree).toMatchSnapshot()
})
