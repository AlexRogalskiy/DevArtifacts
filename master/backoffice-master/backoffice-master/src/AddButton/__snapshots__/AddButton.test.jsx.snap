// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  aria-label="add"
  className="MuiButtonBase-root-28 MuiButton-root-2 MuiButton-fab-19 MuiButton-contained-13 MuiButton-containedSecondary-15 MuiButton-raised-16 MuiButton-raisedSecondary-18 AddButton-button-1"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-3"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-31"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root-38"
  />
</button>
`;
